test/onnx/verify/roialign_half_pixel_verify_test.cpp

```diff
@@ -0,0 +1,61 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2015-2024 Advanced Micro Devices, Inc. All rights reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+#include <migraphx/register_target.hpp>
+#include <migraphx/verify.hpp>
+#include <onnx_test.hpp>
+
+// The half_pixel mode for the ROIAlign op
+TEST_CASE(roialign_half_pixel_verify_test)
+{
+    migraphx::program p = read_onnx("roialign_half_pixel_test.onnx");
+    p.compile(migraphx::make_target("ref"));
+    migraphx::shape s{migraphx::shape::float_type, {2, 2, 4, 3}};
+    std::vector<float> data(2 * 2 * 4 * 3);
+    std::iota(data.begin(), data.end(), 0.f);
+    migraphx::parameter_map pp;
+    pp["x"] = migraphx::argument(s, data.data());
+    pp["y"] = migraphx::argument(s, data.data());
+
+    migraphx::shape srois{migraphx::shape::float_type, {2, 4}};
+    std::vector<float> rois_data = {1.1, 0.73, 1.7, 1.13, 1.1, 0.73, 2.6, 1.13};
+    migraphx::shape sbi{migraphx::shape::int64_type, {2}}; // batch_index
+    std::vector<int64_t> bi_data = {0, 1};
+
+    pp["rois"]      = migraphx::argument(srois, rois_data.data());
+    pp["batch_ind"] = migraphx::argument(sbi, bi_data.data());
+    pp["y"]         = migraphx::argument(s, data.data());
```

Comment:
It's not clear to me whether Y, the output, is even needed for this test.  What role does it play?

Reply:
Likely because `roialign_half_pixel_test.onnx` expects it as input. I am not sure why its configured like that. Maybe it needs an allocated buffer to write the output to?
I am not very familiar with the verify onnx tests, I would check if this is common in other tests also